### PR TITLE
Improve readme

### LINKED ISSUE #336: Improve github readme
the current readme does contain a lot of useful information, but does not state what NEST actually is, nor does it contain all links to relevant resources. also it is questionable to me whether the syntax highlighting for emacs and vim should get such a prominent place. maybe a (very) condensed version of the http://nest-simulator.org/ homepage could be an option?

### PR DESCRIPTION
**Pull request by sanjayankur31**
Should eventually fix #336

### CODE REVIEW

**Comment by apeyser** [comment]
Should we add the current cite link to the README? It will require remembering to update it.

**Comment by sanjayankur31** [comment]
Maybe a release SOP document somewhere that enumerates all steps to be completed? (Something like [this](http://infrastructure.fedoraproject.org/infra/docs/fedora-releases.rst)?)

**Comment by apeyser** [comment]
@sanjayankur31 
@jougs Is there a release doc already?

**Comment by apeyser** [comment]
@heplesser @abigailm 
Would anyone else like to opine on the docs?

File: README.md [hunk 1]
```diff
@@ -1,19 +1,164 @@
...
+  supercomputers.
+- NEST has minimal dependencies. All it really needs is a C++ compiler.
+  Everything else is optional.
+- NEST developers are using agile continuous integration-based workflows in
+  order to maintain high code quality standards for correct and reproducible
+  simulations.
+- NEST has one of the largest and most experienced developer communities of all
+  neural simulators. NEST was first released in 1994 under the name SYNOD and
+  has been extended and improved ever since.
+- NEST is open source software and is licensed under the GNU General Public
```
Comment (heplesser) [inline README.md]: Make the licensing information a section of its own for better visibility.

**Comment by heplesser** [comment]
@sanjayankur31 Excellent! My only suggestion is to make the licensing information more visible, see in-text comment. People will often look for that. Otherwise, :+1:.

**Pull request merged**